src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -196,10 +196,9 @@ trait EntityComponent {
 
       for {
         preExistingEntityRecs <- lookupEntitiesByNames(workspaceContext.workspaceId, entities.map(_.toReference))
-        _ <- deleteEntityAttributes(preExistingEntityRecs)
         savingEntityRecs <- insertNewEntities(workspaceContext, entities, preExistingEntityRecs).map(_ ++ preExistingEntityRecs)
         referencedAndSavingEntityRecs <- lookupNotYetLoadedReferences(workspaceContext, entities, savingEntityRecs).map(_ ++ savingEntityRecs)
-        _ <- insertAttributes(entities, referencedAndSavingEntityRecs)
+        _ <- upsertAttributes(entities, (savingEntityRecs ++ preExistingEntityRecs).distinct, referencedAndSavingEntityRecs)
```

Comment:
have you seen this to be the case it seems like savingEntityRecs is only newly inserted entities and preExistingEntityRecs are ones that already exist so they should be mutually exclusive. Unless it causes an actual problem, I would remove it as it is probably not a cheap operation when upserting thousands of entities.